### PR TITLE
Limit the read length of policyFile

### PR DESCRIPTION
**Pull request by zouyee**
Signed-off-by: Zou Nengren <zouyee1989@gmail.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Limit the read length of policyFile 

**Which issue(s) this PR fixes**:

xref https://github.com/kubernetes/kubernetes/pull/76518 https://github.com/kubernetes/kubernetes/pull/76519

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/scheduler/scheduler.go [hunk 1]
```diff
@@ -44,13 +42,15 @@ import (
...
+	utilio "k8s.io/utils/io"
 )
 
 const (
 	// BindTimeoutSeconds defines the default bind timeout
 	BindTimeoutSeconds = 100
 	// SchedulerError is the reason recorded for events when an error occurs during scheduling a pod.
-	SchedulerError = "SchedulerError"
+	SchedulerError  = "SchedulerError"
+	maxConfigLength = 10 * 1 << 20 // 10MB
```
Comment (mattjmcnaughton) [inline pkg/scheduler/scheduler.go]: Just want to get check that 10MB is an appropriate value - confirming that there is no feasible reason this file would be more than 10MB?

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm

Modulo double checking that 10MB is an appropriate value, this change works for me. I think we have the added benefits of extra protection from `ReadAtMost`, and no downsides.

File: pkg/scheduler/scheduler.go (hunk 1 above)
Comment (zouyee) [reply pkg/scheduler/scheduler.go]: Although the scenario is different, I think this kind of setting is also applicable 
xref: https://github.com/kubernetes/kubernetes/pull/76519#issuecomment-483822126
cc @liggitt

**Review by hex108 — COMMENTED** [review]
Thanks @zouyee for it.

Is there any benefit with this PR? I think #76518 is needed because data of `http readiness/liveness probes` are from users, they might be very large and cause memory problems.  However scheduler's config file is configured by admin and only read one time, there is no previous problem.

**Comment by liggitt** [comment]
> Is there any benefit with this PR?

I agree that protection from abuse by component administrators is not a priority.

**Comment by mattjmcnaughton** [comment]
> > Is there any benefit with this PR?
> 
> I agree that protection from abuse by component administrators is not a priority.

Thanks for your thoughts here @hex108 and @liggitt ! I agree with you both that the risk of cluster admins inflicting damage via this mechanism is quite low.

If its helpful, my justification for the change was that this diff provides another positive example in the code base of reading files with a length limit, which I believe is a practice we generally want to encourage. Additionally, it uses a library function, so the code added is fairly small.

Overall, I see both sides :)

**Pull request closed without merging**